retrofit/src/main/java/retrofit2/KotlinExtensions.kt

```diff
@@ -14,9 +14,51 @@
  * limitations under the License.
  */
 
-// Hide the class from Java consumers.
-@file:JvmName("-KotlinExtensions")
+@file:JvmName("KotlinExtensions")
 
 package retrofit2
 
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
 inline fun <reified T> Retrofit.create(): T = create(T::class.java)
+
+suspend fun <T : Any> Call<T>.await(): T {
+  return suspendCancellableCoroutine { continuation ->
+    continuation.invokeOnCancellation {
+      cancel()
+    }
+    enqueue(object : Callback<T> {
+      override fun onResponse(call: Call<T>, response: Response<T>) {
+        if (response.isSuccessful) {
+          // TODO handle nullability
```

Comment:
Presumably the callers `T` can itself be nullable or not – like the parameter will be `String` or `String?` but to this code it's all just `T`.

Reply:
Which is a real problem in practice since an endpoint declared as `suspend fun foo(): String` can get an HTTP 204 which produces a null body and leaks that into an otherwise non-null type.

We need something like a hand-written Kotlin metadata parser for return type nullability that doesn't pull in any deps or support anything except looking for exactly what we need. That way it's 1 method and not 10,000.